src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1578,97 +1576,46 @@ public function deletePropertyImmediately(string $path): void
     private function deleteProperty(string $path): void
     {
         $this->assertLoggedIn();
-
         $nodePath = PathHelper::getParentPath($path);
-        $nodeId = $this->getSystemIdForNode($nodePath);
-        if (!$nodeId) {
-            // no we really don't know that path
-            throw new ItemNotFoundException('No item found at '.$path);
-        }
-        $this->removePropertiesFromNode($nodeId, [$path]);
+        $propertyName = PathHelper::getNodeName($path);
+        $this->removePropertiesFromNode($nodePath, [$propertyName]);
     }
 
     /**
      * Removes a list of properties from a given node.
      *
-     * @param array<string> $paths Path belonging to that node that should be deleted
+     * @param array<string> $propertiesToDelete Path belonging to that node that should be deleted
      */
-    private function removePropertiesFromNode(string|int $nodeId, array $paths): void
+    private function removePropertiesFromNode(string $nodePath, array $propertiesToDelete): void
     {
+        $nodeId = $this->getSystemIdForNode($nodePath);
+        if (!$nodeId) {
+            // no we really don't know that path
+            throw new ItemNotFoundException("No item found at $nodePath");
+        }
+
         $query = 'SELECT props FROM phpcr_nodes WHERE id = ?';
         $xml = $this->getConnection()->fetchOne($query, [$nodeId]);
 
-        $dom = new \DOMDocument('1.0', 'UTF-8');
-        $dom->loadXML($xml);
-        $xpath = new \DOMXPath($dom);
-
-        $found = false;
-        foreach ($paths as $path) {
-            $propertyName = PathHelper::getNodeName($path);
-            $tablesToDeleteReferencesFrom = [];
-            $deleteBinary = false;
-            foreach ($xpath->query(sprintf('//*[@sv:name="%s"]', $propertyName)) as $propertyNode) {
-                \assert($propertyNode instanceof \DOMElement);
-                $found = true;
-                // would be nice to have the property object to ask for type
-                // but its in state deleted, would mean lots of refactoring
-                if (!$propertyNode->hasAttribute('sv:type')) {
-                    continue;
-                }
-
-                $type = strtolower($propertyNode->getAttribute('sv:type'));
-                switch ($type) {
-                    case 'reference':
-                        $tablesToDeleteReferencesFrom[] = $this->referenceTables[PropertyType::REFERENCE];
-                        break;
-                    case 'weakreference':
-                        $tablesToDeleteReferencesFrom[] = $this->referenceTables[PropertyType::WEAKREFERENCE];
-                        break;
-                    case 'binary':
-                        $deleteBinary = true;
```

Comment:
ah, this is a new feature in 2.x and i did not realize when solving conflicts. we need to figure out how to keep this working

Reply:
i think this means XmlPropsRemover needs to also return the list of binary properties, like it returns the reference properties.

Reply:
sorry, :duck: 

i think i solved it

Reply:
Looks the same way I would have solve it :)